tdp/core/collection.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 TOSIT.IO
+# SPDX-License-Identifier: Apache-2.0
+
+from pathlib import Path
+
+DAG_DIRECTORY_NAME = "tdp_lib_dag"
+ACTION_DIRECTORY_NAME = "playbooks"
+DEFAULT_VARS_DIRECTORY_NAME = "tdp_vars_defaults"
+
+YML_EXTENSION = ".yml"
+
+MANDATORY_DIRECTORIES = [
+    DAG_DIRECTORY_NAME,
+    DEFAULT_VARS_DIRECTORY_NAME,
+    ACTION_DIRECTORY_NAME,
+]
+
+
+class Collection:
+    def __init__(self, path):
+        self._path = Path(path)
+        self._dag_yamls = None
+        self._actions = None
+
+    @staticmethod
+    def from_path(path):
+        path = Path(path).expanduser().resolve()
+        if not path.exists():
+            raise ValueError(f"{path} does not exists")
+        if not path.is_dir():
+            raise ValueError(f"{path} is not a DIRECTORY")
+        for mandatory_DIRECTORY in MANDATORY_DIRECTORIES:
+            mandatory_path = path / mandatory_DIRECTORY
+            if not mandatory_path.exists() or not mandatory_path.is_dir():
+                raise ValueError(
+                    f"{path} does not contain the mandatory DIRECTORY {mandatory_DIRECTORY}",
+                )
+        return Collection(path)
+
+    @property
+    def path(self):
+        return self._path
+
+    @property
+    def name(self):
+        return self._path.name
+
+    @property
+    def dag_DIRECTORY(self):
```

Comment:
Why `DIRECTORY` is in uppercase ?

Reply:
Mistake in renaming folder to directory, fixed